test/functional/system/rounds.js

```diff
@@ -0,0 +1,1584 @@
+									);
+								});
+								done();
+							});
+
+							it('fields vote, blocks_forged_count, blocks_missed_count, isDelegate, virgin should be valid', done => {
+								_.each(delegates, delegate => {
+									// Find accounts that vote for delegate
+									const voters = _.filter(
+										library.genesisblock.block.transactions,
+										transaction => {
+											return (
+												transaction.type === 3 &&
+												transaction.asset.votes.indexOf(
+													`+${delegate.publicKey.toString('hex')}`
+												) !== -1
+											);
+										}
+									);
+
+									// Calculate voters balance for current delegate
+									let voters_balance = '0';
+									_.each(voters, voter => {
+										const balance = _.reduce(
+											library.genesisblock.block.transactions,
+											(balance, acc) => {
+												if (acc.recipientId === voter.senderId) {
+													return new Bignum(balance)
+														.plus(acc.amount)
+														.toString();
+												} else if (acc.senderId === voter.senderId) {
+													return new Bignum(balance)
+														.minus(acc.amount)
+														.toString();
+												}
+												return balance;
+											},
+											'0'
+										);
+										voters_balance = new Bignum(voters_balance)
+											.plus(balance)
+											.toString();
+									});
+
+									expect(delegate.vote).to.equal(voters_balance);
+									expect(delegate.producedBlocks).to.equal(0);
+									expect(delegate.missedBlocks).to.equal(0);
+									expect(delegate.isDelegate).to.equal(1);
+									expect(delegate.u_isDelegate).to.equal(1);
+									expect(delegate.virgin).to.equal(1);
+								});
+								done();
+							});
+						});
+					});
+				});
+
+				describe('mem_accounts (other accounts)', () => {
+					let accounts;
+					let genesisAccounts;
+					let genesisAddress;
+
+					before(() => {
+						// Get genesis accounts address - should be senderId from first transaction
+						genesisAddress =
+							library.genesisblock.block.transactions[0].senderId;
+
+						// Get unique accounts from genesis block
+						genesisAccounts = _.reduce(
+							library.genesisblock.block.transactions,
+							(accounts, transaction) => {
+								if (
+									transaction.senderId &&
+									accounts.indexOf(transaction.senderId) === -1
+								) {
+									accounts.push(transaction.senderId);
+								}
+								if (
+									transaction.recipientId &&
+									accounts.indexOf(transaction.recipientId) === -1
+								) {
+									accounts.push(transaction.recipientId);
+								}
+								return accounts;
+							},
+							[]
+						);
+
+						// Get accounts from database
+						return Queries.getAccounts().then(_accounts => {
+							accounts = _accounts;
+						});
+					});
+
+					it('should be populated', () => {
+						expect(accounts).to.be.an('array');
+						return expect(accounts.length).to.be.above(0);
```

Comment:
We know the number of accounts in geneisBlock, should not we match the exact length.

Reply:
Those tests were designed to support every genesis block (mainnet, custom one, etc.) - that's why expectations are so generic. You can notice that instead of hardcoding everything we detecting genesis account, delegates registrations, etc.